src/serialmonitor/serialportctrl.ts

```diff
@@ -75,51 +107,56 @@ export class SerialPortCtrl {
             const jsonObj = JSON.parse(data.toString())
             this._outputChannel.append(jsonObj["payload"] + "\n");
         });
-        resolve(true);
+        // TODO: add message check to ensure _child spawned without errors
+        resolve();
+        // The spawn event is only supported in node v15+ vscode
+        // this._child.on("spawn", (spawn) => {
+        //     resolve();
+        // });
+
      });
   }
 
-  public sendMessage(text: string): Promise<any> {
+  public sendMessage(text: string): Promise<void> {
     return new Promise((resolve, reject) => {
       if (!text || !this._currentSerialPort || !this.isActive) {
-        resolve(false);
+        resolve();
         return;
       }
 
       this._currentSerialPort.write(text + "\r\n", (error) => {
         if (!error) {
-          resolve(true);
+          resolve();
         } else {
           return reject(error);
         }
       });
     });
   }
 
-  public changePort(newPort: string): Promise<any> {
+  public changePort(newPort: string): Promise<void> {
     return new Promise((resolve, reject) => {
       if (newPort === this._currentPort) {
-        resolve(true);
+        resolve();
         return;
       }
       this._currentPort = newPort;
       if (!this._currentSerialPort || !this.isActive) {
-        resolve(false);
+        resolve();
         return;
       }
       this._currentSerialPort.close((err) => {
         if (err) {
           reject(err);
         } else {
           this._currentSerialPort = null;
-          resolve(true);
+          resolve();
         }
       });
     });
   }
 
-  public stop(): Promise<any> {
-    this._child.stdin.write('{"cmd": "close"}\n');
+  public stop(): Promise<boolean> {
```

Comment:
I don't see uses of `stop()` that use the return value, so you could also convert this method to return a `Promise<void>` like the other methods in this class.

Reply:
serialMonitor.ts uses the return values.